pyquil/tests/test_quantum_computer.py

```diff
@@ -470,3 +470,18 @@ def test_noisy(forest):
     qc = get_qc('1q-qvm', noisy=True)
     result = qc.run_and_measure(p, trials=10000)
     assert result[0].mean() < 1.0
+
+
+def test_run_expects_executable():
+    # https://github.com/rigetti/pyquil/issues/740
+
+    # This test might need some more knowledgeable eyes. Not sure how
+    # to best mock a qpu.
+    qc = get_qc('1q-qvm')
+    qc.qam = QPU(endpoint='tcp://not-needed:00000',
```

Comment:
I think the legit way to mock this would be to mock out `QPU.client.call` which is the rpcq thing that actually communicates with the quantum hardware. Maybe the rpcq repo has some useful test fixtures we could repurpose here?